### ISSUE TITLE
socket unable to connect [err 905]

### ISSUE THREAD

**Issue by LeoStrong23** [jasonacox/tinytuya#526]
hello every one,
this is my first comment in GitHub so excuse me if I wrot in the wrong discussion or other my errors. I'm sorry for my english. I am blind so I think I could'not give you all necessary computer's output, if you need some specifics outputs could you teach me how to get them?

maybe this is the wrong discussion for this problem but I found similar issues here so I thought to write it here.
for a few months I connected my tuya devices to home assistant by local tuya personal component and them wark correctly, but at the begining of july one of them became unavailable and unreachable by local tuya and tinytuya. I surch the internet a lot but I didn't solve the problem, the device continue to be in the same status: unavailable and unreachable.
I have to say that my tuya devices are connected also with google home and alexa and them can see and control the devices, this confused me because I think that local control was harder then cloud control used by them.
when I try to do a tinytuya scan it write this:

TinyTuya (Tuya device scanner) [1.15.1]

[Loaded devices.json - 3 devices]

Scanning on UDP ports 6666 and 6667 and 7000 for devices for 18 seconds...

Scanning... | (18)                                 
Scanning... / (17)                                 
New Broadcast from App at 192.168.1.184 - {'from': 'app', 'ip': '192.168.1.184'}
Scanning... - (17)                                 
...
Stufetta   Product ID = keyaqceqmgp343e7  [Valid Broadcast]:
    Address = 192.168.1.61   Device ID = bffe358be2d9b82cc9sc4o (len:22)  Local Key = redacted Version = 3.3  Type = default, MAC = 18:69:d8:90:70:a6
    Status: {'1': False, '11': 0}
Scanning... - (12)                                 
...
Scanning... | (5)                                 
ventilatore   Product ID = keyjnuy4s3kre7m7  [Valid Broadcast]:
    Address = 192.168.1.187   Device ID = bfb86ec27d3bc50ee9ota1 (len:22)  Local Key = redacted Version = 3.3  Type = default, MAC = d4:a6:51:f9:e1:aa
    Polling 192.168.1.187 Failed: 
Scanning... / (4)                                 
...
Scanning... / (0)                                 
Scan completed in 18.0008 seconds
                    
Scan Complete!  Found 2 devices.
Broadcasted: 2
Versions: 3.3: 2

>> Saving device snapshot data to snapshot.json

if I try to poll the device it respond this:
set_status() result {'Error': 'Network Error: Device Unreachable', 'Err': '905', 'Payload': None}

I found in this section of discussions this post [https://github.com/jasonacox/tinytuya/issues/41](URL), doing that advices in this post to get debugs:
here is the code: `
import tinytuya
import time
import random

tinytuya.set_debug(True)  # use tinytuya.set_debug(True,False) for non-ANSI color terminal

d = tinytuya.BulbDevice("bfb86ec27d3bc50ee9ota1", "192.168.1.187", localKey=redacted)
d.set_version(3.3)

# Show status of device
data = d.status()
print('\nCurrent Status of Bulb: %r' % data)

# Set to full brightness warm white
print('\nWarm White Test')
d.set_white()
time.sleep(1)

# Power Control Test
print('\nPower Control Test')
print('    Turn off lamp')
d.turn_off()
time.sleep(1)
print('    Turn on lamp')
d.turn_on()
time.sleep(1)
`
 I recived the output copied in this File.
[debug.txt](https://github.com/user-attachments/files/16456158/debug.txt)

thank everyone for the support. 
I hope thes informations are sufficient, I  can give you othersif is  necessary.

**Comment by uzlonewolf** [comment]
Hi @LeoStrong23, have you tried removing all power from the device, waiting a few seconds, and then turning it back on?  I've seen devices get confused before and stop responding to local requests and usually a reboot fixes it.

**Comment by LeoStrong23** [comment]
hi @uzlonewolf, at the end I have fixed my problem. was appened that after a rutter  restart the smart switch changed his local key, so tiny tuya can't poll it using the saved local key. I tryed to lounch a new wizard to get news local key but I got a tuya server error saying that my free profile was expired, so I had to apply for an extension of the free tuya developer profile. at that moment I could run a new wizard with out errors and I saw that the local key of the device that had got problems was changed, I put all new informations into local tuya and now all work correctly. for this time I could fix it getting an extention of the tuya iot platform free profile, but I thinc that the nex time I won't apply for another extention period and I don't know how I'll can do it in a free modality. 
Replying to you, I had trieh to do that you say but it not fixed anything, I even tried to delete and reconnect the device to the smart life app, but I thought that if is a problem of localkeys these solution can't work and not influence the local keys.
in any case thank you for the support and I hope that this post can help other people!
